### PR TITLE
Added option to unsubscribe from all channels without knowing their names

### PR DESCRIPTION
**Pull request by marcelosalloum**
Just added a useful function to easily erase the list of subscribed channels

### CODE REVIEW

**Comment by marcelosalloum** [comment]
Fix issue #199

File: Library/PTPusher.m [hunk 1]
```diff
@@ -274,6 +274,20 @@ - (void)subscribeAll
   }
 }
 
+- (void)unsubscribeAllChannels
+{
+  if (channels && channels.count) {
+	NSMutableArray *channelsToUnsubscribe = [[NSMutableArray alloc] init];
+	  for (PTPusherChannel *channel in [channels allValues]) {
+		[channelsToUnsubscribe addObject:channel];
```
Comment (hamchapman) [inline Library/PTPusher.m]: Do we need the intermediate `channelsToUnsubscribe` array?

Can we not just put `[channel unsubscribe];` here?

Comment (marcelosalloum) [inline Library/PTPusher.m]: When you call `[channel unsubscribe]`, that channel will be removed from the `channels` array, which will cause a crash in the loop (the array size cannot change in the middle of the loop)

This crash will only happen when there are at least 2 channels to be unsubscribed.

Please refer to [this Stackoverflow thread](http://stackoverflow.com/questions/111866/best-way-to-remove-from-nsmutablearray-while-iterating) to for more info.

Comment (hamchapman) [inline Library/PTPusher.m]: Ah gotcha - I was being slow!

Comment (marcelosalloum) [inline Library/PTPusher.m]: No problem, you'd need to check the method `- (void)__unsubscribeFromChannel:(PTPusherChannel *)channel` to figure that out.

Comment (lukeredpath) [inline Library/PTPusher.m]: The temporary array is unnecessary here. 'channels' is a dictionary, not an array, and you're already creating a new array of all the subscribed channels when you call 'allValues'. A crash would only occur if you were iterating over the dictionary itself.

Comment (lukeredpath) [inline Library/PTPusher.m]: I'm also don't see the purpose of checking 'channels' (it's created on initialisation) or its count. If the count is zero, then allValues will return an empty array anyway.

Comment (marcelosalloum) [inline Library/PTPusher.m]: Thanks @lukeredpath, I'll fix that and re-submit the PR.

**Comment by hamchapman** [comment]
Hi @marcelosalloum have you had a chance to incorporate Luke's feedback?

**Comment by marcelosalloum** [comment]
Hi @hamchapman, thanks for bringing this up. I've just pushed those changes.

**Comment by pocket7878** [comment]
Any Updates? I want to use it

**Comment by marcelosalloum** [comment]
Hi @pocket7878, it's ready and working. I am just waiting for @hamchapman or @lukeredpath to approve the PR.

**Pull request merged**